src/main/kotlin/us/ihmc/build/IHMCDependenciesExtension.kt

```diff
@@ -1,27 +1,16 @@
 package us.ihmc.build
 
-import groovy.lang.Closure
-import org.gradle.api.Action
 import org.gradle.api.Project
 import org.gradle.api.artifacts.Dependency
-import org.gradle.api.artifacts.ExternalModuleDependency
-import org.gradle.api.artifacts.MinimalExternalModuleDependency
 import org.gradle.api.artifacts.dsl.DependencyHandler
-import org.gradle.api.artifacts.dsl.ExternalModuleDependencyVariantSpec
-import org.gradle.api.provider.Provider
-import org.gradle.api.provider.ProviderConvertible
-import org.gradle.internal.metaobject.DynamicInvokeResult
-import org.gradle.internal.metaobject.MethodAccess
-import org.gradle.internal.metaobject.MethodMixIn
 import org.gradle.kotlin.dsl.extra
 
 open class IHMCDependenciesExtension(private val mainProject: Project,
                                      private val sourceSetKebabCasedName: String,
-                                     private val ihmcBuildExtension: IHMCBuildExtension)
-   : MethodMixIn, // Groovy trick
-     IHMCKotlinDependencyHandlerDelegate()  // Kotlin trick
+                                     private val ihmcBuildExtension: IHMCBuildExtension,
+                                     private val dependencies: DependencyHandler)
+   : DependencyHandler by dependencies  // Allows us to override only the methods we need to
```

Comment:
The `by` keyword does allll the work, here. It allows to override only certain methods, which I believe would not only simplify but provide backwards compatibility.